src/components/remote_control/src/commands/base_command_request.cc

```diff
@@ -0,0 +1,634 @@
+    CheckHMILevel(application_manager::kAllowed, success);
+  }
+}
+
+void BaseCommandRequest::ProcessAccessResponse(
+    const rc_event_engine::Event<application_manager::MessagePtr, std::string>&
+        event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (!app_) {
+    LOG4CXX_ERROR(logger_, "Application doesn't registered!");
+    SendResponse(false, result_codes::kApplicationNotRegistered, "");
+    return;
+  }
+
+  application_manager::Message& hmi_response = *(event.event_message());
+  const application_manager::MessageValidationResult validate_result =
+      service_->ValidateMessageBySchema(hmi_response);
+  LOG4CXX_DEBUG(logger_,
+                "HMI response validation result is " << validate_result);
+
+  if (validate_result !=
+      application_manager::MessageValidationResult::SUCCESS) {
+    SendResponse(
+        false, result_codes::kGenericError, "HMI has sent invalid parameters");
+    return;
+  }
+
+  Json::Value value;
+  Json::Reader reader;
+  reader.parse(event.event_message()->json_message(), value);
+
+  std::string result_code;
+  std::string info;
+  const bool is_succeeded = ParseResultCode(value, result_code, info);
+
+  bool is_allowed = false;
+  if (is_succeeded) {
+    if (IsMember(value[kResult], message_params::kAllowed) &&
+        value[kResult][message_params::kAllowed].isBool()) {
+      is_allowed = value[kResult][message_params::kAllowed].asBool();
+    }
+
+    const std::string module = ModuleType(msg_json_);
+
+    // Check the actual User's answer.
+    LOG4CXX_DEBUG(logger_,
+                  "Setting allowed access for " << app_->app_id() << " for "
+                                                << module);
+    service_->SetAccess(app_->app_id(), module, is_allowed);
+    CheckHMILevel(application_manager::kManual, is_succeeded);
+
+    if (is_allowed) {
+      rc_module_.resource_allocation_manager().ForceAcquireResource(
+          module, app_->app_id());
+
+      Execute();  // run child's logic
+    } else {
+      rc_module_.resource_allocation_manager().OnDriverDisallowed(
+          module, app_->app_id());
+
+      SendResponse(
+          false,
+          result_codes::kRejected,
+          "The resource is in use and the driver disallows this remote "
+          "control RPC");
+    }
+  } else {
+    SendResponse(false, result_code.c_str(), info);
+  }
+}
+
+void BaseCommandRequest::CheckHMILevel(application_manager::TypeAccess access,
+                                       bool user_consented) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  switch (access) {
+    case application_manager::kAllowed:
+      if (user_consented) {
+        if (app_->hmi_level() == mobile_apis::HMILevel::eType::HMI_NONE) {
+          LOG4CXX_DEBUG(logger_,
+                        "RSDL functionality for "
+                            << app_->name().c_str()
+                            << " is auto allowed; setting BACKGROUND level.");
+          service_->ChangeNotifyHMILevel(
```

Comment:
why ChangeNotifyHMILevel during CheckHMLevel?
not sure about the logic here

Reply:
@LuxoftAKutsan Should this be removed?

Reply:
@JackLivio method removed as unused